models/review.go

```diff
@@ -255,6 +265,11 @@ func SubmitReview(doer *User, issue *Issue, reviewType ReviewType, content strin
 		return nil, nil, err
 	}
 
+	// Only reviewers latest review shall count as "official", so existing reviews needs to be cleared
+	if _, err := sess.Exec("UPDATE `review` SET official=? WHERE issue_id=? AND reviewer_id=?", false, issue.ID, doer.ID); err != nil {
```

Comment:
I didn't consider this, you are right only approving/rejecting review shall count for the official attribute. I think l have fixed that now.

It is correct that clearing old "official" reviews is not needed to get the latest reviews of each reviewer. That query still returns reviews from each reviewer, even those that are not official (i.e in the "approval whitelist"). However the reason you wanted this was to efficiently calculate number of approvals, and if you loook at `GetGrantedApprovalsCount` it sums it up directly. So therefore it is necessary to clear old approvals. Also I think it can be handed if we implement _block at "Changes requested"_, we can easily do a direct SQL query to see if there are any official review with status Rejected/Changes requested.